src/Symfony/Component/Cache/Adapter/MemcacheAdapter.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Rob Frawley 2nd <rmf@src.run>
+ */
+class MemcacheAdapter extends AbstractAdapter
+{
+    use MemcacheAdapterTrait;
+
+    /**
+     * Construct adapter by passing a \Memcache instance and an optional namespace and default cache entry ttl.
+     *
+     * @param \Memcache   $client
+     * @param string|null $namespace
+     * @param int         $defaultLifetime
+     */
+    public function __construct(\Memcache $client, $namespace = '', $defaultLifetime = 0)
+    {
+        parent::__construct($namespace, $defaultLifetime);
+        $this->client = $client;
+    }
+
+    /**
+     * Factory creation method that provides an instance of this adapter with a\Memcache client instantiated and setup.
+     *
+     * Valid DSN values include the following:
+     *  - memcache://localhost                  : Specifies only the host (defaults used for port and weight)
+     *  - memcache://example.com:1234           : Specifies host and port (defaults weight)
+     *  - memcache://example.com:1234?weight=50 : Specifies host, port, and weight (no defaults used)
+     *
+     * @param string|null $dsn
+     *
+     * @return MemcacheAdapter
+     */
+    public static function create($dsn = null)
+    {
+        if (!extension_loaded('memcache') || !version_compare(phpversion('memcache'), '3.0.8', '>')) {
```

Comment:
The behavior of this extension on version 3.0.8 seems completely different from the behavior of 3.0.9 - IDK why that would be the case with what should be a patch release, but tests fail with any version below 3.0.9.